static/app/utils/profiling/profile/sampledProfile.tsx

```diff
@@ -0,0 +1,119 @@
+import {lastOfArray} from 'sentry/utils';
+import {CallTreeNode} from 'sentry/utils/profiling/callTreeNode';
+
+import {Frame} from './../frame';
+import {Profile} from './profile';
+import {createFrameIndex} from './utils';
+
+export class SampledProfile extends Profile {
+  static FromProfile(sampled: Profiling.SampledProfile): Profile {
+    const {startValue, endValue, samples, weights} = sampled;
+
+    const sampledProfile = new SampledProfile(
+      endValue - startValue,
+      startValue,
+      endValue,
+      sampled.name,
+      sampled.unit
+    );
+
+    const frameIndex = createFrameIndex(sampled.shared.frames);
+
+    if (samples.length !== weights.length) {
+      throw new Error(
+        `Expected samples.length (${samples.length}) to equal weights.length (${weights.length})`
+      );
+    }
+
+    for (let i = 0; i < samples.length; i++) {
+      const stack = samples[i];
+      const weight = weights[i];
+
+      sampledProfile.appendSampleWithWeight(
+        stack.map(n => frameIndex[n]),
```

Comment:
Can `frameIndex[n]` here be undefined? Should this guard against it?

Reply:
It should not be, that would mean that the trace is corrupt, would be fine to throw here, I’ll add it